src/Helpers.fs

```diff
@@ -73,59 +63,78 @@ module Process =
-        else
-            let prms = seq { yield location; yield! cmd'} |> ResizeArray
-            node.childProcess.spawn(linuxCmd, prms, options)
-
-    let spawnInDir location linuxCmd (cmd : string) =
-        let cmd' = splitArgs cmd |> ResizeArray
-
-        let options =
-            createObj [
-                "cwd" ==> (node.path.dirname location)
-            ]
-        if isWin () || linuxCmd = "" then
-
-            node.childProcess.spawn(location, cmd', options)
-        else
-            let prms = seq { yield location; yield! cmd'} |> ResizeArray
-            node.childProcess.spawn(linuxCmd, prms, options)
-
-    let spawnWithShell location linuxCmd (cmd : string) =
-        let cmd' =
-            if isWin() then
-                splitArgs cmd
-                |> Seq.map (fun c -> if c.Contains " " then sprintf "\"%s\"" c else c )
-                |> ResizeArray
-            else
-                splitArgs cmd
-                |> Seq.toArray
-                |> ResizeArray
-
-        if isWin () || linuxCmd = "" then
-            window.createTerminal("F# Application", location, U2.Case1 cmd')
+    let private createCommandAndParameters windowsCommand linuxCommand args = 
+        if isWin () || linuxCommand = "" then
+            windowsCommand, args
         else
-            let prms = seq { yield location; yield! cmd'} |> ResizeArray |> U2.Case1
-            window.createTerminal("F# Application", linuxCmd, prms)
-
+            let prms = seq { yield windowsCommand; yield! args } |> ResizeArray
+            linuxCommand, prms
+
+    /// <summary>
+    /// spawns a child process in a directory, using a prefix command on linux if required
+    /// </summary>
+    /// <param name="dir">the directory to run the command in</param>
+    /// <param name="windowsCommand">the 'base' command to execute</param>
+    /// <param name="linuxCommand">if on linux, this runner command will be used as the command and the windowsCommand will be passed as the first parameter</param>
+    /// <param name="args">an array of additional CLI args</param>
+    /// <returns></returns>
+    let spawnInDir dir windowsCommand linuxCommand args =
+        let options = createEmpty<ExecOptions>
+        options.cwd <- Some dir
+        let command, args = createCommandAndParameters windowsCommand linuxCommand args
+        node.childProcess.spawn(command, args, options)
+
+    /// <summary>
+    /// Spawns a child process with a prefix command on linux if required.
+    /// </summary>
+    /// <param name="windowsCommand">the 'base' command to execute</param>
+    /// <param name="linuxCommand">if on linux, this runner command will be used as the command and the windowsCommand will be passed as the first parameter</param>
+    /// <param name="args">an array of additional CLI args</param>
+    /// <returns>a node child process</returns>
+    let spawn windowsCommand linuxCommand (args: ResizeArray<string>) =
```

Comment:
I wonder if we still need windows command vs linux command? Back in the days, we used that for windows .exe calls vs mono

Reply:
I think you're right here. every tool not uses `dotnet` as the runner, and the other places we do linux-specific things are entirely different calls to this API, because we're calling entirely different commands that don't follow the windows-prefix rules.